std/typecons.d

```diff
@@ -2333,7 +2333,24 @@ non-null. Calling $(D nullify) can nullify it again.
 
 Practically $(D Nullable!T) stores a $(D T) and a $(D bool).
  */
-struct Nullable(T)
+template Nullable(T)
+{
+    alias Nullable = NullableWithExtraField!T;
+}
+
+
+/**
+Defines a value paired with a distinctive "null" state that denotes
+the absence of a value. If default constructed, a $(D Nullable!T)
+object starts in the null state. Assigning it renders it non-null.
+Calling $(D nullify) can nullify it again.
+
+Practically $(D Nullable!T) stores a $(D T) and a $(D bool). Note that being
+nullified is not the same as assigning the value to a `null` value.
+For example, if `T` is a class, assigning the value to `null` does not nullify
+this struct.
+ */
+struct NullableWithExtraField(T)
```

Comment:
Hmm this is a new public symbol. Can't we move this into the eponymous template and thus make it private for now (and also save on template instantiations)?

Reply:
Lol, I just finished editing the description to explain why I think it should be "public" :)

Reply:
Hmm, but what's the use case for this? Is this just to simply the deprecation period?
If we really have to do so, can't we use an additional flag like `Nullable!(ExtraField.Yes, T, nullValue)`? Adding new public symbols shouldn't be done lightly. Once added, we never get rid of them.